### PR TITLE
fix: Sidestep batch loader to fix SaleRegistrationConnection

### PR DESCRIPTION
**Pull request by jo-rs**
This PR relates to [PURCHASE-2185].

It sidesteps an issue that was raised after merging [this](https://github.com/artsy/metaphysics/pull/2894) PR in which the batch loader was causing the result to have an unexpected shape. Once merged, the part pertaining to `force` can be brought back.

Note: The PR mentioned above was working locally, but not on Staging due to the `ENABLE_RESOLVER_BATCHING` environment variable not being set locally.

Done alongside @zephraph 

cc @artsy/purchase-devs 

[PURCHASE-2185]: https://artsyproduct.atlassian.net/browse/PURCHASE-2185

### CODE REVIEW

**Comment by zephraph** [comment]
Yeah, there's multiple issues here that raise some complexity. Largely the batch-loaders don't support multiple arguments at this point. If we had better typing, part of this issue would've been communicated sooner. In general batched loaders shouldn't return a different type if it goes through batching or not... but currently it is, which I consider to be a bug. There's probably a lot more here that would just take some dedicated thought on how the API behaves vs what we'd actually like to see. 

With that said, for this issue, it's not worth digging into all those things. Given we don't really want to batch anything here _and_ that we need headers, it's better to just use the specified loader.

File: _schemaV2.graphql [hunk 1]
```diff
@@ -8874,6 +8876,7 @@ type Profile {
   isPubliclyVisible: Boolean
   isPublished: Boolean
   name: String
+  profileArtistsLayout: String
```
Comment (starsirius) [inline _schemaV2.graphql:8879]: Are these schema changes intentional?

**Comment by sepans** [comment]
Nice catch! 🚀 

I once tried to flip the ENV variables one by one to see if I can reproduce the problem locally but I guess I missed `ENABLE_RESOLVER_BATCHING`

File: _schemaV2.graphql (hunk 1 above)
Comment (zephraph) [reply _schemaV2.graphql:8879]: I suspect they're not

File: src/schema/v2/me/sale_registrations.ts [hunk 2]
```diff
@@ -38,24 +38,21 @@ export const SaleRegistrationConnection: GraphQLFieldConfig<
...
-    ) as any)) as BodyAndHeaders
+    const { body: sales, headers } = (await (salesLoaderWithHeaders({
+      is_auction,
+      live,
+      published,
+      sort,
+      page,
+      size,
+      total_count: true,
+    }) as any)) as BodyAndHeaders
```
Comment (damassi) [inline src/schema/v2/me/sale_registrations.ts:55]: This form is pretty confusing to read, can we type it like:

```tsx
const response: BodyAndHeaders = await salesLoaderWithHeaders(...)
const { body, headers } = response
...
```

File: _schemaV2.graphql (hunk 1 above)
Comment (jo-rs) [reply _schemaV2.graphql:8879]: Umm, no 🤔 
I wonder why that changed. Good catch, @starsirius!

**Review by jo-rs — COMMENTED** [review]
~~Umm why didn't the revert push any changed?🤔~~

File: src/schema/v2/me/sale_registrations.ts (hunk 2 above)
Comment (jo-rs) [reply src/schema/v2/me/sale_registrations.ts:55]: Yessir!

File: _schemaV2.graphql (hunk 1 above)
Comment (jo-rs) [reply _schemaV2.graphql:8879]: I just reverted the file but they got automatically changed again on the pre-commit hook

Comment (zephraph) [reply _schemaV2.graphql:8879]: Hmmm, maybe you need to rebase?

Comment (jo-rs) [reply _schemaV2.graphql]: Super messy git state 🤦 
Should be fine now. Sorry, folks.

**Review by starsirius — APPROVED** [review]
🎉

**Review by zephraph — APPROVED** [review]

**Pull request merged**